tools/ejections/cmd/main.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/geth"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/eth"
+	"github.com/Layr-Labs/eigenda/disperser/dataapi"
+	"github.com/Layr-Labs/eigenda/disperser/dataapi/subgraph"
+	"github.com/Layr-Labs/eigenda/tools/ejections"
+	"github.com/Layr-Labs/eigenda/tools/ejections/flags"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/urfave/cli"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	version   = "1.0.0"
+	gitCommit = ""
+	gitDate   = ""
+)
+
+type EjectionTransaction struct {
+	BlockNumber           uint64            `json:"block_number"`
+	BlockTimestamp        string            `json:"block_timestamp"`
+	TransactionHash       string            `json:"transaction_hash"`
+	QuorumStakePercentage map[uint8]float64 `json:"stake_percentage"`
+	QuorumEjections       map[uint8]uint8   `json:"ejections"`
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Version = fmt.Sprintf("%s,%s,%s", version, gitCommit, gitDate)
+	app.Name = "ejections report"
+	app.Description = "operator ejections report"
+	app.Usage = ""
+	app.Flags = flags.Flags
+	app.Action = RunScan
+	if err := app.Run(os.Args); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func RunScan(ctx *cli.Context) error {
+	config, err := ejections.NewConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	logger, err := common.NewLogger(config.LoggerConfig)
+	if err != nil {
+		return err
+	}
+
+	client, err := geth.NewMultiHomingClient(config.EthClientConfig, gethcommon.Address{}, logger)
+	if err != nil {
+		return err
+	}
+
+	tx, err := eth.NewReader(logger, client, config.BLSOperatorStateRetrieverAddr, config.EigenDAServiceManagerAddr)
+	if err != nil {
+		return err
+	}
+
+	chainState := eth.NewChainState(tx, client)
+	if chainState == nil {
+		return errors.New("failed to create chain state")
+	}
+	subgraphApi := subgraph.NewApi(config.SubgraphEndpoint, config.SubgraphEndpoint)
+	subgraphClient := dataapi.NewSubgraphClient(subgraphApi, logger)
+
+	ejections, err := subgraphClient.QueryOperatorEjectionsForTimeWindow(context.Background(), int32(config.Days), config.OperatorId, config.First, config.Skip)
```

Comment:
looks like this section of code is highly similar to the codeblock above. Create a new function?